mongorestore/mongorestore_test.go

```diff
@@ -1084,3 +1084,71 @@ func TestSkipSystemCollections(t *testing.T) {
 		})
 	})
 }
+
+// TestSkipStartAndAbortIndexBuild asserts that all "startIndexBuild" and "abortIndexBuild" oplog
+// entries are skipped when restoring the oplog.
+func TestSkipStartAndAbortIndexBuild(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+	ctx := context.Background()
+
+	sessionProvider, _, err := testutil.GetBareSessionProvider()
+	if err != nil {
+		t.Fatalf("No cluster available: %v", err)
+	}
+	session, err := sessionProvider.GetSession()
+	if err != nil {
+		t.Fatalf("No client available")
+	}
+
+	if ok, _ := sessionProvider.IsReplicaSet(); !ok {
+		t.SkipNow()
+	}
+
+	sessionProvider.GetNodeType()
+
+	Convey("With a test MongoRestore instance", t, func() {
+		testdb := session.Database("test")
+
+		// Drop the collection to clean up resources
+		defer testdb.Collection("skip_index_entries").Drop(ctx)
+
+		args := []string{
+			DirectoryOption, "testdata/oplog_ignore_index",
+			OplogReplayOption,
+			DropOption,
+		}
+
+		currentTS := uint32(time.Now().UTC().Unix())
+
+		restore, err := getRestoreWithArgs(args...)
+		So(err, ShouldBeNil)
+
+		if restore.serverVersion.GTE(db.Version{4, 4, 0}) {
+			// Run mongorestore
+			result := restore.Restore()
+			So(result.Err, ShouldBeNil)
+
+			Convey("RestoreOplog() should ignore all startIndexBuild and abortIndexBuild oplog entries", func() {
+				queryObj := bson.D{
+					{"$and",
+						bson.A{
+							bson.D{{"ts", bson.M{"$gte": primitive.Timestamp{T: currentTS, I: 1}}}},
+							bson.D{{"$or", bson.A{
+								bson.D{{"o.startIndexBuild", primitive.Regex{Pattern: "skip_index_entries"}}},
```

Comment:
This test is not ideal since it's just checking if we didn't create startIndexBuild or abortIndexBuild in the target oplog. But what should be really checked is such entries are ignored.
For example, if we convert the`startIndexBuild` to `createIndexes` in the target this test won't be able to find that. I am not sure if you can create a bson file with startIndexBuild and abortIndexBuild alone and prove we don't create any oplog on the destination?

Reply:
I've modified `oplog.bson` so that it only has `startIndexBuild` and `abortIndexBuild` entries, and have modified the test to ensure that nothing is added to the oplog after running mongorestore.

Reply:
I just noticed that the test is failing on evergreen despite passing locally. Currently trying to figure out why the number of documents in `db.local.oplog.rs` > 0 even though the logs say "applied 0 oplog entries".